test/e2e/tests/show-hex-data.spec.js

```diff
@@ -0,0 +1,115 @@
+  detailsTab: { text: 'Details', tag: 'button' },
+  containerContent: '.confirm-page-container-content',
+  confirmButton: {
+    text: 'Confirm',
+    tag: 'button',
+    css: '[data-testid="page-container-footer-next"]',
+  },
+};
+
+const inputData = {
+  recipientAddress: '0x985c30949c92df7a0bd42e0f3e3d539ece98db24',
+  hexDataText: '0xabc',
+};
+
+// Function to click elements in sequence
+async function clickElementsInSequence(driver, clickSelectors) {
+  for (const selector of clickSelectors) {
+    await driver.waitForSelector(selector);
+    await driver.clickElement(selector);
+  }
+}
+
+// Function to perform the hex data toggle
+async function toggleHexData(driver) {
+  const sequence = [
+    selectors.accountOptionsMenu,
+    selectors.settingsDiv,
+    selectors.advancedDiv,
+    selectors.hexDataToggle,
+  ];
+
+  await clickElementsInSequence(driver, sequence);
+}
+
+// Function to click on the app logo
+async function clickOnLogo(driver) {
+  await driver.clickElement(selectors.appHeaderLogo);
+}
+
+// Function to send a transaction and verify hex data text area appears.
+async function sendTransactionAndVerifyHexData(driver) {
+  await driver.clickElement(selectors.ethOverviewSend);
+  await driver.fill(selectors.ensInput, inputData.recipientAddress);
+  await driver.fill(selectors.quantity, 1);
+  await driver.fill(selectors.hexDataInput, inputData.hexDataText);
+  await driver.clickElement(selectors.nextPageButton);
+  await driver.clickElement(selectors.hexButton);
+}
+
+// Main test suite
+describe('Check the toggle for hex data', function () {
+  let ganacheOptions;
+
+  it('Setting the hex data toggle and verify that the textbox appears', async function () {
+    ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: convertToHexValue(25000000000000000000),
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver, ganacheServer }) => {
+        await logInWithBalanceValidation(driver, ganacheServer);
+        await toggleHexData(driver);
+        await clickOnLogo(driver);
+        await sendTransactionAndVerifyHexData(driver);
+
+        // Verify hex data in the container content
+        const pageContentContainer = await driver.findElement(
+          selectors.containerContent,
+        );
+        const pageContentContainerText = await pageContentContainer.getText();
+        assert.equal(
+          pageContentContainerText.includes(inputData.hexDataText),
+          true,
+          'Hex data is incorrect',
+        );
+
+        // Click on the 'Details' tab
+        await driver.clickElement(selectors.detailsTab);
+
+        // Click on the 'Confirm' button
+        await driver.clickElement(selectors.confirmButton);
+      },
```

Comment:
Is there any assertion after clicking the confirm button we needed here?

Reply:
The test focuses on toggling and verifying the presence of Hex Data. The provided code is designed to validate the Hex Data on the confirmation page, with the subsequent step being the click on the confirm button. While it's possible to place the assert statement after the transaction is finalized, the task specifies that it concludes with the button click.

assert.equal(
          pageContentContainerText.includes(inputData.hexDataText),
          true,
          'Hex data is incorrect',
        );

Reply:
@hjetpoluru to clarify this part of the task description:

> The test should then click next, and confirm that the expected hex data is shown on the Data tab of the confirm screen. The test does not need to take further steps after that.

That should probably say: "The test should then click next, and _assert_ that the expected hex data is shown on the Data tab of the confirm screen. The test does not need to take further steps after that."

So after asserting that the hex data is present, the test does not need to click any further buttons